Here's also the specific logs for which this PR is based around. :sweat_smile: 

```c
In file included from /build/981zjvzg8d0nix5lw9h2s1x8l3ga48dj-source/test/test_calculator.c:2:
/build/981zjvzg8d0nix5lw9h2s1x8l3ga48dj-source/extern/mtest/include/mtest.h:60:37: warning: named variadic macros are a GNU extension [-Wvaria>
#define TEST_CASE(test_name, command...)                                                                                \
                                    ^
[ 83%] Linking C executable project1
/build/981zjvzg8d0nix5lw9h2s1x8l3ga48dj-source/test/test_calculator.c:75:1: warning: comparison of integers of different signs: 'int' and 'uns>
MAIN_RUN_TESTS(no_op_square_root_of_negative_numbers,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/981zjvzg8d0nix5lw9h2s1x8l3ga48dj-source/extern/mtest/include/mtest.h:91:44: note: expanded from macro 'MAIN_RUN_TESTS'
        for (int i = 0; token != NULL && i < sizeof(test_functions)/sizeof(int(*)(void)); ++i) {                        \
                                         ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/981zjvzg8d0nix5lw9h2s1x8l3ga48dj-source/test/test_calculator.c:75:1: warning: comparison of integers of different signs: 'int' and 'uns>
MAIN_RUN_TESTS(no_op_square_root_of_negative_numbers,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/981zjvzg8d0nix5lw9h2s1x8l3ga48dj-source/extern/mtest/include/mtest.h:104:12: note: expanded from macro 'MAIN_RUN_TESTS'
    } else RUN_TESTS(__VA_ARGS__);                                                                                      \
           ^~~~~~~~~~~~~~~~~~~~~~
/build/981zjvzg8d0nix5lw9h2s1x8l3ga48dj-source/extern/mtest/include/mtest.h:72:23: note: expanded from macro 'RUN_TESTS'
    for (int i = 0; i < sizeof(test_functions)/sizeof(int(*)(void)); ++i)                                               \
                    ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 warnings generated.
```

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,5 +1,7 @@
 add_executable(test_mtest test_mtest.c)
 target_link_libraries(test_mtest mtest)
+# Avoid relying on hidden compiler behavior when compiling test_mtest.
+target_compile_options(test_mtest PRIVATE -Wall -Wextra -Wpedantic)
 
 # Automatically discover test cases in test_mtest and add them to CTest.
 discover_tests(test_mtest)

